### PR TITLE
Typed Repository and typed Query

### PR DESCRIPTION
**Pull request by dennishendriksen**

### CODE REVIEW

File: molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchRepository.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ public EntityMetaData getEntityMetaData()
 	@Override
 	public void rebuildIndex()
 	{
-		Iterable<Entity> entities = elasticSearchService.search(new QueryImpl(this), getEntityMetaData());
+		Iterable<Entity> entities = elasticSearchService.search(new QueryImpl<>(this), getEntityMetaData());
```
Comment (jjettenn) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchRepository.java]: Did you forgot this <Entity>?

File: molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchService.java [hunk 2]
```diff
@@ -354,7 +354,7 @@ public long count(Query q, EntityMetaData entityMetaData)
 			if (hasMapping(transactionId, entityMetaData))
 			{
 				// count added entities in transaction index
-				Query countAddedQ = q != null ? new QueryImpl(q) : new QueryImpl();
+				Query<Entity> countAddedQ = q != null ? new QueryImpl<>(q) : new QueryImpl<>();
```
Comment (jjettenn) [inline molgenis-data-elasticsearch/src/main/java/org/molgenis/data/elasticsearch/ElasticsearchService.java]: QueryImpl<>  is not QueryImpl<Entity>?

File: molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/ElasticsearchServiceTest.java [hunk 3]
```diff
@@ -170,7 +170,7 @@ public void search()
 		DefaultEntityMetaData entityMetaData = new DefaultEntityMetaData("entity");
 		entityMetaData.setBackend(ElasticsearchRepositoryCollection.NAME);
 		entityMetaData.addAttribute(idAttrName, ROLE_ID).setDataType(MolgenisFieldTypes.INT);
-		Query q = new QueryImpl();
+		Query<Entity> q = new QueryImpl<>();
```
Comment (jjettenn) [inline molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/ElasticsearchServiceTest.java]: new QueryImpl<>() --> new QueryImpl<Entity>()

Comment (jjettenn) [inline molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/ElasticsearchServiceTest.java]: new QueryImpl<Entity>()

File: molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/request/QueryGeneratorTest.java [hunk 4]
```diff
@@ -673,7 +673,7 @@ public void generateOneQueryRuleInXref_Entities()
 		ref2.set(idAttributeName, "id2");
 
 		Iterable<Object> values = Arrays.<Object> asList(ref0, ref1, ref2);
-		Query q = new QueryImpl().in(xrefAttributeName, values);
```
Comment (jjettenn) [inline molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/request/QueryGeneratorTest.java]: new QueryImpl<>() --> new QueryImpl<Entity>() multi places

File: molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/request/SourceFilteringGeneratorTest.java [hunk 5]
```diff
@@ -78,7 +79,7 @@ public void generateElasticsearchBackendFetchSubFetchWithSubFetch()
 		String refAttr0Name = "refAttr0";
 		String refRefAttr0Name = "refRefAttr0";
 		when(entityMeta.getBackend()).thenReturn(ElasticsearchRepositoryCollection.NAME);
-		Query q = new QueryImpl().fetch(new Fetch().field(attr0Name).field(attr1Name,
+		Query<Entity> q = new QueryImpl<>().fetch(new Fetch().field(attr0Name).field(attr1Name,
```
Comment (jjettenn) [inline molgenis-data-elasticsearch/src/test/java/org/molgenis/data/elasticsearch/request/SourceFilteringGeneratorTest.java]: Entity

File: molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/controller/MappingServiceController.java [hunk 6]
```diff
@@ -291,7 +291,7 @@ public AttributeMappingValidationReport validateAttributeMapping(
 			}
 		};
 
-		long total = dataService.count(sourceEntityName, new QueryImpl());
+		long total = dataService.count(sourceEntityName, new QueryImpl<>());
```
Comment (jjettenn) [inline molgenis-data-mapper/src/main/java/org/molgenis/data/mapper/controller/MappingServiceController.java]: Entity

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataSearchServiceImpl.java [hunk 7]
```diff
@@ -44,7 +44,7 @@ public MetaDataSearchServiceImpl(DataService dataService, MetaDataService metaDa
 		else
 		{
 			// Search in packages
-			Query q = new QueryImpl().search(searchTerm);
+			Query<Entity> q = new QueryImpl<>().search(searchTerm);
 			// for (Entity packageEntity : dataService.findAllAsIterable(PackageMetaData.ENTITY_NAME, q))
```
Comment (jjettenn) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataSearchServiceImpl.java]: remove this comment?

**Comment by jjettenn** [comment]
Entity is missing in some places, this is done deliberately?

**Comment by jjettenn** [comment]
PR too big some diff's cannot be opened

**Comment by dennishendriksen** [comment]
@jjettenn Regarding your generic comments please read [this article](http://docs.oracle.com/javase/7/docs/technotes/guides/language/type-inference-generic-instance-creation.html). Could you review your comments afterwards and let me know if there are still changes you would like to see?

File: molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataSearchServiceImpl.java (hunk 7 above)
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/data/meta/MetaDataSearchServiceImpl.java]: This comment already existed. The class was automatically updated due to a refactoring.

**Comment by jjettenn** [comment]
Can you please solve branch conflicts?

**Comment by dennishendriksen** [comment]
Hoi Jonathan,

Dit had ik eerder deze week ook met een ander PR en is een bug in GitHub. Een workaround is het sluiten en heropenen van het PR. Kan ik dit doen?

Groeten,
Dennis

________________________________

**Comment by jjettenn** [comment]
Ja

**Pull request merged**